docs/protocol/src/setup/contributions.md

```diff
@@ -0,0 +1,191 @@
+
+We break the CRS described [previously](./groth16_recap.md) into two parts:
+
+First, we have:
+
+- $[\alpha]_1, [\beta]_1, [\beta]_2$
+
+- $\displaystyle \left[x^i\right]_1\quad (i \in [0, \ldots, 2d - 2])$
+
+- $\displaystyle \left[x^i\right]_2\quad (i \in [0, \ldots, d - 1])$
+
+- $\displaystyle \left[\alpha x^i\right]_1\quad (i \in [0, \ldots, d - 1])$
+
+- $\displaystyle \left[\beta x^i\right]_1\quad (i \in [0, \ldots, d - 1])$
+
+
+Second, we have:
+
+- $[\delta]_1, [\delta]_2$
+
+- $\displaystyle \left[\frac{1}{\delta} p^{\alpha, \beta}_i(x)\right]_1\quad (i \geq s)$
+
+- $\displaystyle \left[\frac{t(x)}{\delta} x^i \right]_1\quad (i \in [0, \ldots, d - 2])$
+
+We split the ceremony into two phases, to calculate the first and second
+part of the CRS, respectively.
+The general idea in each ceremony is that the secret values of interest
+(e.g. $\alpha, x$ etc.) are shared multiplicatively, as
+$\alpha_1 \cdot \alpha_2 \ldots$, with each party having
+one of the shares.
+Because of this structure, given the current value of the CRS elements
+in a given phase, it's possible for a new party to add their contribution.
+For example, in the first phase, one can multiply each element
+by some combination $\alpha^{d_1} \cdot \beta^{d_2} \cdot x^{d_3}$,
+depending on the element,
+to get a new CRS element.
+
+Each contribution will come with a proof of knowledge for the new secret
+values contributed, which can also partially attest to how these secret
+values were used.
+However, this is not enough to guarantee that the resulting elements
+are a valid CRS: for this, we have a consistency check
+allowing us to check that the elements in a given phase
+have the correct internal structure.
+
+Each party can thus contribute one after the other, until
+enough contributions have been gathered through that phase.
+
+In order to link phase 1 and phase 2,
+we use the fact that with $\delta = 1$, the CRS elements of phase 2
+are linear combinations of those in phase 1.
+If we consider $t(x)x^i$, with $i$ up to $d - 2$,
+the largest monomial we'll find is $2d - 2$, since $t$ has degree at most $d$.
+In the first phase, we calculated these powers of $x$, and so can
+calculate these values by linear combination.
+We can do the same for:
+$$
+p_i^{\alpha, \beta}(x) = \alpha u_i(x) + \beta v_i(x) + w_i(x)
+$$
+since we have access to $\alpha x^i$ and $\beta x^i$ for sufficiently
+high degrees.
+
+## Phase 1
+
+Assuming we have the CRS elements of phase 1, a contribution involves
+fresh random scalars $\hat{\alpha}, \hat{\beta}, \hat{x}$, and produces
+the following elements:
+
+- $\hat{\alpha} \cdot [\alpha]_1, \hat{\beta} \cdot [\beta]_1, \hat{\beta} \cdot [\beta]_2$
+
+- $\hat{x}^i \cdot [x^i]_1\quad (i \in [0, \ldots, 2d - 2])$
+- $\hat{x}^i \cdot [x^i]_2\quad (i \in [0, \ldots, 2d - 2])$
+- $\hat{\alpha}\hat{x}^i \cdot [\alpha x^i]_1\quad (i \in [0, \ldots, d - 1])$
+- $\hat{\beta}\hat{x}^i \cdot [\beta x^i]_1\quad (i \in [0, \ldots, d - 1])$
```

Comment:
Doesn't need to be in this PR but at some point we should document where we deviate from KMSV21 (e.g. when I was looking at this I was comparing with Fig. 6 in the paper)